packages/walletconnect-v2/src/index.ts

```diff
@@ -135,8 +135,22 @@ export class WalletConnect extends Connector {
 
     if (provider.session) {
       if (!desiredChainId || desiredChainId === provider.chainId) return
-      if (!this.chains.includes(desiredChainId)) {
-        throw new Error(`Cannot activate chain (${desiredChainId}) that was not included in initial options.chains.`)
+      /*
+       * Unfortunately, WalletConnect does not provide a better/cleaner way to access connected chains.
+       * We have to work with the list of accounts in format: `eip155:${chain}:${address}`.
+       */
+      const isConnectedToDesiredChain = provider.session.namespaces.eip155.accounts.some((account) =>
+        account.startsWith(`eip155:${desiredChainId}:`)
+      )
+      if (!isConnectedToDesiredChain) {
+        if ((this.options.optionalChains || []).includes(desiredChainId)) {
+          throw new Error(
+            `Cannot activate an optional chain (${desiredChainId}). This error is expected, because there is no guarantee that the user is connected to a given optional chain. Either handle this error in your code or remove the chain from the list of optional chains.`
```

Comment:
Most of this is integration instruction - I think it belongs in a doc-comment, not in a thrown error.

Reply:
Arguably, you are correct, but on the other hand, if this error comes up in someone else's Sentry, I think it may make debugging easier. Don't think there's any harm in providing additional debugging information attached to an error.